Finish instruction implementation

halaney opened the pull request:
Finished all the instructions to be implemented. Note that this was done
quite lazily. Definitely better "rust" ways to do this versus the quick
and dirty c++ that I had implemented this with originally.

Removed all the weird ownership stuff I added previously. No need for
shared ownership in this situation. Just complicates things.

Added a few unit tests (not nearly enough!) and a helper function
to create test roms more easily.

Commit:
Finish instruction implementation

Finished all the instructions to be implemented. Note that this was done
quite lazily. Definitely better "rust" ways to do this versus the quick
and dirty c++ that I had implemented this with originally.

Removed all the weird ownership stuff I added previously. No need for
shared ownership in this situation. Just complicates things.

Added a few unit tests (not nearly enough!) and a helper function
to create test roms more easily.

halaney commented on src/lib.rs:
```diff
@@ -73,11 +57,20 @@ impl Chip8 {
+    // Clear the screen
+    fn clear_screen(&mut self) {
+        for (_i, row) in self.screen.iter_mut().enumerate() {
+            for column in row.iter_mut() {
```
column should probably be pixel here

MattCatz commented on src/lib.rs:
```diff
@@ -1,55 +1,38 @@
+    // The screen is 64x32 pixels, i.e 64 wide 32 tall. There are 32 rows,
+    // 64 columns. I try to stick to that notation here but it should probably
+    // be changed to screen[width][height] like most graphics apis seem to be
+    pub screen: [[bool; 64]; 32],
```
What us the bool for?

MattCatz commented on src/lib.rs:
```diff
@@ -240,59 +233,53 @@ impl Chip8 {
-                            }
-                            mScreen[(currentHeight+y)%32][(currentWidth+x)%64] ^= pixel;
+                        // Check for collision
+                        if self.screen[((current_height + y) % 32) as usize]
```
You use this (Current_height + y) % 32 several times maybe just pull it out and do let y_offset = (Current_height + y) % 32.

Commit:
Make a few small tweaks

Notably, this fixes a bug where the instruction register
(used by programs to reference sprites) was being incorrectly
set causing sprites to be loaded to the screen incorrectly.

Also updates documentation and clarifies a few variables.
